### PR TITLE
ISSUE 2708 - Revamping - Container revision details

### LINKED ISSUE #2708: Revamping - Container revision details
### Description
When clicking an item in the containers list it should display a more detailed item providing a revision list and the options to add a new revision or to upload the file for the first time

### Goals
- [x] Users should be able to click on an item in the list and the details of the container will appear
- [x] If the container is empty the details will show an "upload file" button and be able to upload a new file
- [x] If the container is not empty, the list of revisions should appear with a "add new revision" button to upload a new file and create a new revision

### Tasks
- [x] Create a specific list control with the selected style (maybe using a "selected theme" where all the primary/secondary colors  are set as the specification of selected?)
- [x] Create a specific list item with the selected style
- [x] Create the trail on the left of the list, this should be prepare to for n revisions. ask @george-hadfield  for one or two revisions
- [x] Create a new saga/selectors/redux in v5 for fetching the revisions
- [x] Add tests for this saga

### Related Resources
revisions trail:
image

mockup: https://www.figma.com/proto/djgxvRKOaMMyutswn5yywU/Teamspace?node-id=825%3A3916&scaling=min-zoom&page-id=821%3A1258

### PR DESCRIPTION
**Pull request by m1chalkubiak**
This fixes #2708

Test url: https://issue-2708.dev.3drepo.io/v5/dashboard/carmen/80e710a0-05bc-11ec-8edf-ab0f7cc84da8/t/containers

### CODE REVIEW

**Comment by m1chalkubiak** [comment]
@sanmont3drepo I moved the information about the revisions to a separate redux store, this has simplified the logic. Please check if you see anything else that You think would be worth changing.

**Comment by m1chalkubiak** [comment]
@george-hadfield Can you direct me where I can find guidelines or designs for how you would like to views should behave on smaller resolutions?

**Comment by george-hadfield** [comment]
> @george-hadfield Can you direct me where I can find guidelines or designs for how you would like to views should behave on smaller resolutions?

This will be address early next week when I return from holiday - but possibly before if I have time.

**Review by sanmont3drepo — CHANGES_REQUESTED** [review]

File: frontend/test/revisions/revisions.redux.spec.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {
+	INITIAL_STATE,
+	reducer as revisionsReducer,
+	RevisionsActions,
+} from '@/v5/store/revisions/revisions.redux';
+import { times } from 'lodash';
+import { revisionsMockFactory } from '@/v5/store/revisions/revisions.fixtures';
```
[the remark this answers is not in the document]
Comment (m1chalkubiak) [reply frontend/test/revisions/revisions.redux.spec.ts]: sure, done.

**Review by george-hadfield — APPROVED** [review]
Everything is fine from my perspective apart from the responsive at smaller screen sizes which I will be providing tomorrow.

**Pull request merged**